include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -800,4 +800,21 @@ def VectorizeReturnOp : ArcOp<"vectorize.return", [
   let assemblyFormat = "operands attr-dict `:` qualified(type(operands))";
 }
 
+def AssertOp : ArcOp<"assert"> {
+  let summary = "assert that a properity holds.";
+  let description = [{
+    Asserts that an operation holds at runtime, it takes a single boolean
+    as an input, and an error message.
+    If the property holds (= `true`) the operation has no effect. Otherwise, the
+    execution aborts and provide an error message to the user.
+    ```mlir
+    assert %proberty, "Some error message in case of failure!"
+    ```
+  }];
+  // I thought of making the string message optional.
+  // Also should it take clock as parameters?
```

Comment:
In the long run, a single clock is not enough since the asserted property could reason across multiple clock domains. But for a start it's probably easiest to add a single clock operand (also easier than having no clock).